packages/components/icons/src/inline-svg.tsx

```diff
@@ -0,0 +1,97 @@
+import type { Props } from './templates/icon.styles';
+
+import React from 'react';
+import { oneLine } from 'common-tags';
+import { useTheme } from '@emotion/react';
+import { getColor, getSizeDimensions } from './templates/icon.styles';
+
+type InlineSvgProps = Props & {
+  as: 'image' | 'iframe';
+  data: string;
+};
+
+const defaultProps: Pick<InlineSvgProps, 'as'> = {
+  as: 'image',
+};
+
+const fillColorRe = /\sfill="#[A-Fa-f0-9]+"/g;
+const dimensionWidthRe = /\swidth="[\w]+"/g;
+const dimensionHeightRe = /\sheight="[\w]+"/g;
+
+const InlineSvgAsImage = (props: InlineSvgProps) => {
+  const theme = useTheme();
+  const fillColor = getColor(props.color, theme);
+  const dimensions = getSizeDimensions(props.size);
+
+  // Manually patch the SVG to apply color and sizes, as we can't use css styles.
+  // NOTE the leading whitespace is on purpose, as we define the RegExp with `\s`.
+  const styledSvg = props.data
+    .replace(fillColorRe, ` fill="${fillColor}"`)
+    .replace(dimensionWidthRe, ` width="${dimensions.width}"`)
+    .replace(dimensionHeightRe, ` height="${dimensions.height}"`);
+
+  // Render SVG as base64 data URI, to prevent any sort of embedded scripts to be executed.
+  return (
+    <img
+      {...dimensions}
+      src={`data:image/svg+xml;base64,${btoa(styledSvg)}`}
+      alt=""
+    />
+  );
+};
+InlineSvgAsImage.displayName = 'InlineSvgAsImage';
+
+const InlineSvgAsIFrame = (props: InlineSvgProps) => {
+  const theme = useTheme();
+  const fillColor = getColor(props.color, theme);
+  const dimensions = getSizeDimensions(props.size);
+  const iframeWidth = props.size === 'scale' ? '100%' : dimensions.width;
+  const iframeHeight = props.size === 'scale' ? '100%' : dimensions.height;
+
+  const iFrameStyles = oneLine`<style>
+    html, body {
+      margin: 0;
+      padding: 0;
+      display: flex;
+      overflow: hidden;
+      width: ${iframeWidth};
+      height: ${iframeHeight};
+    }
+    svg {
+      width: ${dimensions.width};
+      height: ${dimensions.height};
+    }
+    * {
+      fill: ${fillColor};
+    }
+  </style>`;
+
+  // Render SVG within an sandboxed iframe, to prevent any sort of embedded scripts to be executed.
+  return (
+    // eslint-disable-next-line jsx-a11y/iframe-has-title
+    <iframe
```

Comment:
Can you share some research you did driving you decision to use an `iframe` instead of sanitising the SVG using e.g. `dompurify` which is about `17.7 KB` (non GZIP)?

Reply:
I did look into various methods and libraries to sanitize the SVG (not dompurify admittedly) and most of the "guides/tutorials" were recommending those two approaches, mostly as a secure way of preventing any sort of XSS.

However, now that I look at dompurify it does indeed seem to do the job.

I'll take another look at this, thanks for the tip.

The important thing is that there is no risk of XSS.

Reply:
Yes, that should be our goal. I was just wondering if an iframe tackles all the issues or leaves some vectors open. I know that DOMPurify is used by many large companies to do exactly this.

Generally I would think that we given a user can upload an SVG somewhere:

1. Use `dompurity` on the API to sanitize also anything we save
2. Additionally sanitize anything we show which isn't coming from us

Somehow maybe the 2. point might not be strictly needed if we don't inject anything uploaded into the DOM before not having sanitized it in the API. 🤔